lib/ansible/plugins/filter/core.py

```diff
@@ -465,13 +465,38 @@ def b64decode(string):
     return to_text(base64.b64decode(to_bytes(string, errors='surrogate_or_strict')))
 
 
+def flatten(mylist, levels=None):
+
+    ret = []
+    for element in mylist:
+        if element in (None, 'None', 'null'):
+            # ignore undefined items
+            break
+        elif isinstance(element, MutableSequence):
+            if levels:
+                levels = int(levels) - 1
+                if levels >= 0:
+                    ret.extend(flatten(element, levels=levels))
+                else:
+                    ret.append(element)
+            elif levels is None:
+                ret.extend(flatten(element))
+            else:
+                break
```

Comment:
This seems wrong to me.

Consider `levels=1`.  You decrement that to `levels=0`, then the `if levels >= 0:` branch gets taken and you recursively call `flatten(element, levels=0)`.  Here you find a nested subelement that is a list, so you check `if levels:` which is False, then you check `if levels is None`, which is also False, so you break, truncating `mylist` at the 1st item that happens to be a list.

Reply:
hmm, not sure how to check that initial input in levels is non 0

Reply:
@mgedmin should be fixed now